llama-index-core/llama_index/core/node_parser/relational/base_element.py

```diff
@@ -342,5 +346,10 @@ def get_nodes_from_elements(self, elements: List[Element]) -> List[BaseNode]:
             nodes.extend(cur_text_nodes)
             cur_text_el_buffer = []
 
-        # remove empty nodes
+        # remove empty nodes and keep node original metadata inherited from parent nodes
+        for node in nodes:
+            if metadata_inherited:
+                for key in metadata_inherited:
+                    if key not in node.metadata:
```

Comment:
A slightly better way is node.metadata.update(inherited_metadata) -- but not a huge deal